R/estimate_ascertainment.R

```diff
@@ -84,7 +77,14 @@ estimate_ascertainment <- function(data,
   )
   checkmate::assert_int(burn_in, lower = 0)
   checkmate::assert_date(max_date, null.ok = TRUE)
-  checkmate::assert_class(epidist, "epidist", null.ok = TRUE)
+
+  # check delay_density and run over short sequence to test output
+  stopifnot(
+    "`delay_density` must be a distribution density function with 1 argument\\
+    evaluating density at a vector of values and returning a numeric vector.\\
+    E.g. function(x) stats::dgamma(shape = 5, scale = 1, x = x)" =
+      checkmate::test_function(delay_density, nargs = 1, null.ok = TRUE)
+  )
```

Comment:
Just flagging the first occurrence but this should be updated throughout the codebase: the `\\` shorthand for new lines is a glue feature and doesn't work with base functions:

``` r
stopifnot("This is a very long message\\
          with line breaks" = FALSE)
#> Error: This is a very long message\
#>           with line breaks
```

<sup>Created on 2023-10-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Fixed in https://github.com/epiverse-trace/cfr/commit/0117f2a3885e69a040d28c091c64a2d425b2db2f